pkgs/by-name/ma/maelstrom-clj/package.nix

```diff
@@ -28,7 +28,8 @@ stdenv.mkDerivation rec {
     makeWrapper ${jdk}/bin/java $out/bin/maelstrom \
       --add-flags -Djava.awt.headless=true \
       --add-flags "-jar $out/lib/maelstrom.jar" \
-      --set PATH ${lib.makeBinPath runtimeDependencies}
+      --set PATH ${lib.makeBinPath runtimeDependencies} \
+      --suffix PATH : $PATH
```

Comment:
Are you sure this does what you wanted? Did it work for you locally?

I would have thought
`--suffix : PATH ${lib.makeBinPath runtimeDependencies}`
was the solution.
(you can also use `--prefix` instead of `--suffix` if you want to make the runtime deps take precedence)

Reply:
Thanks for help me. Yeah, i'm sure, i tested, and i follow the principle as is described [here](https://nixos.org/manual/nixpkgs/unstable/#fun-makeWrapper) for the suffix flag of makeWrapper.

> --prefix essentially hard-codes dependencies into the wrapper. They cannot be overridden without rebuilding the package.
> If dependencies should be resolved at runtime, use --suffix to append fallback values to PATH.

Reply:
It's important to note that `$PATH` is resolved while still inside the build environment.
This means that PATH first get overwritten by `--set` to be `runtimeDependencies` and then all the all the build-time tools found in `$PATH`, such as `gcc`, `grep`, `make` and all the `coreutils` binaries get appended to it.

I'm not sure how it's possibly working for you. It should not be able to find anything from the runtime PATH.

Do you have an example where it didn't work before, but works after this change?

Reply:
Thanks to clarify. I see how it was possible to work now. Maybe the PATH used was from my session at build time 😢, my bad.
Do you see any another way to invoke the CLI using user runtime PATH var? I don't `--suffix : PATH ${lib.makeBinPath runtimeDependencies}` will work as expected btw.

Reply:
It will work.

Reply:
sure i'll test and update the PR. Thanks for help me.